### ISSUE TITLE
TypeScript convention: evaluate constructor private/public parameters.  aka parameter properties

### ISSUE THREAD

**Issue by samreid** [phetsims/chipper#1191]
See https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/

**Comment by samreid** [comment]
It seems this is up for the dev team to decide how to proceed.  My recommendation is to leave the rule "off", that is, allow developers to use any form of parameter properties as needed.

**Comment by samreid** [comment]
During https://github.com/phetsims/studio/issues/272 I saw a good candidate for using parameter properties, so I committed it and referenced above.

Pros: Reduce boilerplate and duplication. Clear what is happening. Parameter value, property declaration and assignment no longer have to be split up into 3 separate places.
Cons: Multiple ways to declare class properties may make it harder to skim/search/document things (since we would have 2 ways to do that).

Still, that commit was a positive experience and I think we should make an effort to learn and embrace this pattern as an option.
